fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/visitor/NondeterministicFunctionCollector.java

```diff
@@ -43,7 +45,14 @@ public Void visit(Plan plan, FunctionCollectContext collectContext) {
         }
         for (Expression expression : expressions) {
             Set<Expression> nondeterministicFunctions =
-                    expression.collect(expr -> !((ExpressionTrait) expr).isDeterministic());
+                    expression.collect(expr -> {
+                        boolean containsNondeterministic = !((ExpressionTrait) expr).isDeterministic();
+                        if (!collectContext.getCollectExpressionTypes().isEmpty()) {
+                            containsNondeterministic &= collectContext.getCollectExpressionTypes().stream()
+                                    .anyMatch(type -> type.isAssignableFrom(expr.getClass()));
```

Comment:
why need `type.isAssignableFrom(expr.getClass())`

Reply:
have removed this